packages/@aws-cdk/aws-kinesisanalytics-flink/lib/application.ts

```diff
@@ -851,15 +906,24 @@ class Import extends ApplicationBase {
   public readonly applicationName: string;
   public readonly applicationArn: string;
 
-  constructor(scope: Construct, id: string, attrs: { applicationArn: string, applicationName: string }) {
+  constructor(scope: Construct, id: string, attrs: { applicationArn: string, securityGroups?: ec2.ISecurityGroup[] }) {
     super(scope, id);
 
     // Imported applications have no associated role or grantPrincipal
     this.grantPrincipal = new iam.UnknownPrincipal({ resource: this });
     this.role = undefined;
 
     this.applicationArn = attrs.applicationArn;
-    this.applicationName = attrs.applicationName;
+    const applicationName = core.Stack.of(scope).splitArn(attrs.applicationArn, core.ArnFormat.SLASH_RESOURCE_NAME).resourceName;
+    if (!applicationName) {
+      throw new Error(`applicationArn for fromApplicationArn (${attrs.applicationArn}) must include resource name`);
+    }
+    this.applicationName = applicationName;
```

Comment:
why are we getting this from the arn now? This worked when it was in the `Import` class, because that ARN was always an actual ARN. This ARN is almost certainly not an actual string, it's instead a Token, which means we can't validate this at synth time

Reply:
Before this code was in the `Application.fromApplicationName` static method and now it's in the `Import` constructor. So I think this will still work. I think this is the same code path as before but with a slight refactoring.

Also I have [tests for `fromApplicationName` and `applicationArn`](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-kinesisanalytics-flink/test/application.test.ts#L588-L613) that didn't change in this PR. Let me know if there is some edge case I'm missing.